packages/auth/src/app.ts

```diff
@@ -391,6 +382,44 @@ export class App {
     this.introspectionServer = koa.listen(port)
   }
 
+  public async startInteractionServer(port: number | string): Promise<void> {
+    const koa = await this.createKoaServer()
+
+    const router = new Router<DefaultState, AppContext>()
+    router.use(bodyParser())
+
+    const openApi = await this.container.use('openApi')
+    const interactionRoutes = await this.container.use('interactionRoutes')
+
+    // Grant accept/reject
+    router.post<DefaultState, ChooseContext>(
+      '/grant/:id/:nonce/:choice',
+      createValidatorMiddleware<ChooseContext>(openApi.idpSpec, {
```

Comment:
I would consider any route that pulls from the Open Payments IDP spec to be part of this move. So like lines 304 to 332.

Reply:
You mean to also have following routes on this port:
* Start interaction
* Finish interaction
* Grant lookup

Normally, this issue was created due to security concerns regarding accepting and rejecting grants on behalf of xy user. So I would normally create separate ticket and also a separate PR.

But from my understanding Grant lookup should be publicly accessible right? Also for interaction start agreed? Because if interaction start is not publicly accessible then that would mean that you are not redirected to IDP site. So ASE would need to implement also some own startInteraction route and call then start interaction on Rafiki Auth ... Would that also mean to change the start interaction API to return url where the ASE should redirect user? Basically return 301 with IDP_URL ?

Same question for finish ... what finish does right now is that it finished the interaction and redirects the browser session to client callback. So we would also need to change this API right? And also from my memory I remember that something needs to be passed to the finish interaction route? Something that is in cookie right?

IMHO that would be quite an overhaul so if we would go down this road I would like to have separate ticket for this improvement. What do you think @sabineschaller @mkurapov ?

Reply:
Grant lookup does not need to be publicly accessible - at least, not the one that's implemented here. GNAP doesn't specify that grants should be publically queried and the purpose that endpoint in particular is so that the IDP hosting the interaction can lookup the grant - there isn't any wider audience that endpoint is intended for.

I'll walk back the suggestion to move the start/finish interaction, those are meant to be public.